p2p/peer_behaviour.go

```diff
@@ -0,0 +1,137 @@
+package p2p
+
+import (
+	"errors"
+	"sync"
+)
+
+// Type of error behaviour a peer can perform.
+type ErrorPeerBehaviour int
+
+const (
+	ErrorPeerBehaviourUnknown = iota
```

Comment:
Will we ever want to construct the unknown version of this behaviour? Can we make it impossible somehow to construct it?

One way to somewhat encode this is to do `iota + 1` for the first known one, so that the zero value for `int` is accidentally interpreted as a supported value.